### DISCUSSION TITLE
Use only one GPU with Python

### DISCUSSION THREAD

**Discussion by Emvlt** [CERN/TIGRE#329] [Q&A] [answered]
**What I want to do:**
Use only one GPU when running TIGRE AND calling its by it's ID (my two GPUs have the same name)

**What I understand:**
When instantiating a reconstruction algorithm, gpuids=None by defauls and I don't see that it can be changed by passing gpuids as a kwarg.

**Example:**
iterative_recon_alg.py  
- no gpuids in keyworded argumens
- gpuids = None (line 160) which by default grabs all gpus 

OS : Linux
Python version : 3.6
CUDA : 10.1

**Comment by tsadakane** [comment, marked as answer]
@Emvlt Hi.
If you want to use the first GPU, 
```
gpuids = gpu.getGpuIds()
gpuids.devices = [0]  # [1] if you use the second one.
proj = tigre.Ax(head, geo, angles, gpuids=gpuids)
```

You can find more info in `example.py` and `d22_ListGPUs.py`

**Reply by Emvlt** [reply]
Thanks a lot!
I tried to pass the GPU id as an integer to Ax but code memory error, didn't think about passing the object :)

**Comment by epapoutsellis** [comment]
I also followed the above python scripts for an fdk reconstruction but when I run

```python
gpuids = gpu.getGpuIds()
img = tigre.algorithms.fdk(stack_img, geo, angles, gpuids = gpuids)
```
it does not seem (nvidia-smi, task manager) that **both my two** gpus are utilized.

OS : Windows
Python version : 3.9
CUDA : 11.0

Any ideas?

**Reply by Emvlt** [reply]
Have you tried adding 
`gpuids.devices = [0]` or `gpuids.devices = [1]` for device selection?

**Reply by AnderBiguri** [reply]
I think this was because for very large images, most time is used in filtering in FDK, which is not GPU-based yet (albeit #423 exists, and I think its working), @Emvlt
